### PR TITLE
Change links to use HTTPs

### PR DESCRIPTION
**Pull request by zschallz**
- Changed template links to use HTTPS
- Added a test to ensure http:// isn't referenced anywhere in the templates unless it's whitelisted.

### CODE REVIEW

File: templates/index.html [hunk 1]
```diff
@@ -182,7 +182,6 @@ <h3>Discord</h3>
         <p>
           <a class="button-small" href="https://github.com/jmhobbs/terminal-parrot">Terminal</a>
           <a class="button-small" href="https://github.com/Shrugs/partyparrot">Text-To-Parrots</a>
-          <a class="button-small" href="http://countryparrots.com/">Country Parrots</a>
```
Comment (zschallz) [inline templates/index.html]: This is a dead link, so I removed it.

File: templates/index.html [hunk 2]
```diff
@@ -211,15 +210,15 @@ <h3>Discord</h3>
         </p>
 
         <p>
-          <a class="button-small" href="http://kakaporecovery.org.nz/sirocco/">Donate To Keep Them Partying</a>
+          <a class="button-small" href="https://www.doc.govt.nz/kakapo-recovery">Donate To Keep Them Partying</a>
```
Comment (zschallz) [inline templates/index.html]: Domain `kakaporecovery.org.nz` doesn't seem to have an https certificate, but the link redirects to `www.doc.govt.nz` anyway, which does.

File: test/templates.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        });
+
+    templates.forEach(function(file) {
+      let fileContents = fs.readFileSync(__dirname + '/../templates/' + file);
+      
+      let httpLinks = 
+        fileContents
+        .toString()
+        .match(/(http:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*))/gi);
+      httpLinks = httpLinks ? httpLinks : [];
```
Comment (zschallz) [inline test/templates.js]: `.match()` returns null, but it's easier to deal with an empty array.

**Comment by jmhobbs** [comment]
Top notch stuff, thanks for the effort on this!

File: test/templates.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+var fs = require('fs'),
+  assert = require('chai').assert,
+  path = require('path');
+
+describe("HTML templates", function() {
+  it("should not contain un-whitelisted http://", function() {
+    const whiteList = ['http://www.gtfoparrot.com/'];
```
Comment (zschallz) [inline test/templates.js]: No choice here- gtfoparrot is hosted on google storage and has no HTTPS cert.

**Pull request merged**